packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientOptionsProviderTests.cs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Generator.CSharp.ClientModel.Providers;
+using Microsoft.Generator.CSharp.Input;
+using Microsoft.Generator.CSharp.Primitives;
+using Moq;
+using NUnit.Framework;
+
+namespace Microsoft.Generator.CSharp.ClientModel.Tests.Providers
+{
+    public class ClientOptionsProviderTests
+    {
+        private const string ApiVersionsCategory = "WithApiVersions";
+        private readonly Mock<InputLibrary> _mockInputLibrary = new(MockHelpers.ConfigFilePath);
+        private readonly List<string> _apiVersions = ["1.0", "2.0"];
+
+        [SetUp]
+        public void SetUp()
+        {
+            var categories = TestContext.CurrentContext.Test?.Properties["Category"];
+            bool containsApiVersions = categories?.Contains(ApiVersionsCategory) ?? false;
+
+            // Load the mock plugin with or without api versions
+            if (containsApiVersions)
+            {
+                InputEnumTypeValue[] enumValues = new InputEnumTypeValue[_apiVersions.Count];
+                for (var i = 0; i < _apiVersions.Count; i++)
+                {
+                    enumValues[i] = new InputEnumTypeValue(_apiVersions[i], _apiVersions[i], null);
+                }
+                var inputEnum = new InputEnumType(
+                    "ServiceVersion",
+                    string.Empty,
+                    null,
+                    null,
+                    "ServiceVersion description",
+                    InputModelTypeUsage.ApiVersionEnum,
+                    InputPrimitiveType.Int64,
+                    enumValues,
+                    false);
+                var mockInputNs = new Mock<InputNamespace>("ns", _apiVersions, new List<InputEnumType> { inputEnum }, Array.Empty<InputModelType>(), Array.Empty<InputClient>(), new InputAuth());
```

Comment:
all of this mocking should go into the mockhelper and you should pass lambdas into the LoadMockPlugin to assign different behaviors to the different components.